Migrate from Google search to MkDocs.

d0ugal opened the pull request:
**NOTE: [This is dependant on MkDocs pull request 457.](https://github.com/mkdocs/mkdocs/pull/457)**

Putting this up for review, but it is likely that it can't be merged for a while. All of these commits except the last are applicable to the latest MkDocs release ([but there is a bug](https://github.com/mkdocs/mkdocs/issues/461) that needs a point release). I can break the cleanup commits at some point if it helps as they are not really related to search at all - sorry!

This is what the current work looks like at the moment.

![Search screenshot](https://dl.dropboxusercontent.com/u/14255217/Screenshots/Screenshot%202015-04-21-07.55.13.Google-chrome.png)

tomchristie commented:
Looks great!

d0ugal commented:
I've done a few updates so that the theme only loads the search index when the form is submitted. This is a bit nicer, as before it was loading 1.1mb of JSON on any page on load.

Other than improvements in style, which I'm not great at, I can't think of anything else that needs to be done here. If anyone has time to test this out and give feedback I'd be grateful. Ideally it would be good to get feedback before I land the change in MkDocs. https://github.com/mkdocs/mkdocs/issues/457

d0ugal commented on docs_theme/js/theme.js:
```diff
@@ -1,5 +1,35 @@
     $('pre code').parent().addClass('prettyprint well');
 
+    $(document).on("submit", "#mkdocs_search_modal form", function (e) {
+        $("#mkdocs-search-results").html("Searching...")
```
Need a `;` here.

jpadilla commented:
@d0ugal I've been checking this out. Truly great and exciting work you've been pushing out with mkdocs!

tomchristie commented:
From my POV I'm happy with the screenshot @ @jpadilla's review.
I'm ok for this to go in once it's in MkDocs.

d0ugal commented:
Cool. Merged in MkDocs! I'll come back to this once we get closer to a release, I'm thinking it shouldn't be too long away.

d0ugal commented:
MkDocs 0.13 has now been released. I think this probably just needs some testing to check it all still works okay. I'll try and do this soon but initially focusing on upgrading ReadTheDocs which is stuck on MkDocs 0.11 (and is thus providing me with many confused users).

tomchristie commented:
Looking super lovely. :sparkles:

d0ugal commented:
I've rebased and updated this PR. As far as I can tell, it should be ready. My only concern would be cross browser support, I have only tested the latest FireFox and Chrome.

tomchristie commented:
Looks great - may as well merge this in now.
I did notice that when first using a page the searching appeared to only initiate on `<enter>` in the text box. After pressing enter once the searching then worked automatically on text changes.
No idea if the issue is specific to `mkdocs serve` or if it'd also appear on live.

d0ugal commented:
To avoid loading the index on page load, I think it starts the download when you first search. So it might just be a delay as it downloads the JSON. I forget exactly how I set it up in the theme, I'll check.
